cmd/ufs/ufs.go

```diff
@@ -6,15 +6,34 @@ import (
 	"log"
 	"net"
 
-	"github.com/rminnich/ninep/stub"
 	"github.com/rminnich/ninep/filesystem"
+	"github.com/rminnich/ninep/stub"
+	"os"
+	"runtime/pprof"
 )
 
 var (
-	ntype = flag.String("ntype", "tcp4", "Default network type")
-	naddr = flag.String("addr", ":5640", "Network address")
+	ntype      = flag.String("ntype", "tcp4", "Default network type")
+	naddr      = flag.String("addr", ":5640", "Network address")
+	cpuprofile = flag.String("cpuprofile", "", "write cpu profile to file")
 )
 
+func profile(s string, i ...interface{}) {
+	switch s {
+	case "Starting readNetPackets":
+		// start profile
+
+		f, err := os.Create(*cpuprofile)
+		if err != nil {
+			log.Fatal(err)
+		}
+		pprof.StartCPUProfile(f)
+	case "Stop readNetPackets":
+		//stop
+		pprof.StopCPUProfile()
+	}
+}
```

Comment:
Yes with Trace getting called all over the place you'd think it would be in the cpu profile, but all I ever got were 280 bytes of mostly NULL.  That's not the latest code btw (it gets stranger), but yes I can do it differently.